src/Generator.cpp

```diff
@@ -1,11 +1,75 @@
+
 #include <cmath>
 #include <fstream>
 #include <set>
+#include <tuple>
+
+#include "llvm/Support/raw_os_ostream.h"  /// for LLVM and STL I/O bridge
+#include "llvm/Support/YAMLTraits.h"      /// for YAML underpinnings
+#include "runtime/HalideRuntime.h"        /// for halide_type_code_t
 
 #include "Generator.h"
 #include "Outputs.h"
 #include "Simplify.h"
 
+namespace Halide {
+namespace Internal {
+
+/// Forward-declare Halide::Internal::YamlEmitter, for use
+/// in both the forward-declaration of its corresponding
+/// llvm::yaml::MappingTraits<T> specialization, and so that
+/// it may be declared as a YAML sequence vector (q.v the
+/// macro LLVM_YAML_IS_SEQUENCE_VECTOR() invocation sub.)
+class YamlEmitter;
+
+} /// namespace Internal
+} /// namespace Halide
+
+namespace llvm {
+namespace yaml {
+
+/// Make the internal GeneratorParam, GeneratorInput,
```

Comment:
Style nit: the /// style of comment isn't really used anywhere else in Halide. Prefer the existing block-comment style we use elsewhere.